Quantum-Control-Applications/NV-Centers/Confocal-Setup/raw_adc_traces.py

```diff
@@ -10,26 +11,51 @@
 ###################
 # The QUA program #
 ###################
+n_avg = 1000
 
 with program() as TimeTagging_calibration:
-    raw_adc_st = declare_stream(adc_trace=True)
-    play("laser_ON", "AOM")
-    measure("long_readout", "SPCM", raw_adc_st)
-    wait(1000, "SPCM")
+    n = declare(int)
+    adc_st = declare_stream(adc_trace=True)
+    with for_(n, 0, n < n_avg, n + 1):
+        play("laser_ON", "AOM")
+        measure("long_readout", "SPCM", adc_st)
+        wait(1000, "SPCM")
 
     with stream_processing():
-        raw_adc_st.input1().save("raw_adc")
+        # Will save average:
+        adc_st.input1().average().save("adc1")
+        # Will save only last run:
+        adc_st.input1().save("adc1_single_run")
 
 #####################################
 #  Open Communication with the QOP  #
 #####################################
 qmm = QuantumMachinesManager(qop_ip)
-
+# Open Quantum Machine
 qm = qmm.open_qm(config)
-
+# Execute program
 job = qm.execute(TimeTagging_calibration)
-job.result_handles.wait_for_all_values()
+# create a handle to get results
 res_handles = job.result_handles
-raw_data = res_handles.get("raw_adc").fetch_all()
-plt.plot(raw_data)
-plt.title("ADC Trace Check ADCs saturation and define threshold")
+# Wait untill the program is done
+res_handles.wait_for_all_values()
```

Comment:
You're using the "normal" method because there is only one?

Reply:
From what I understood, the idea is that we continue to teach customers how to use standard QUA and fetch data "normally" in the first scripts and then show then the tools progressively. This is the reason why the first scripts don't use the py_qua_tools but the others do